### PR TITLE
Switch to yup validation error handling rather than custom errors

### LINKED ISSUE #2123: Implement more flexible validation error messages on the frontend
### Description

Currently, we implement our own validation messages to accompany `yup` validation errors. This solution is inflexible&mdash;it only allows one error message per input field.

### Proposed solution 

Use built-in `yup` error handling to allow for dynamic error messages based on what validation actually failed.

### PR DESCRIPTION
**Pull request by nickscialli-usds**
## Related Issue or Background Info

- Closes #2123

## Changes Proposed

- This is intended to be a no-op for the user but enables more custom error messages in the future
- To accomplish the no-op, this PR includes some redundant error message strings that, in the future, can be customized for the specific failure type (e.g., a field is missing vs. incorrectly formatted)

## Additional Information

- There are some gnarly `useEffect` hook calls to make sure to re-render when language changes. Long term we may want to consider a form system that better integrates with validation and translation libraries like react-hook-form

### UI
- [n/a] Any changes to the UI/UX are approved by design 
- [ ] Any new or updated content (e.g. error messages) are approved by design 

### Testing
- [n/a] Includes a summary of what a code reviewer should verify

### Changes are Backwards Compatible
- [n/a] Database changes are submitted as a separate PR
  - [n/a] Any new tables that do not contain PII are accompanied by a GRANT SELECT to the no-PHI user
  - [n/a] Any changes to tables that have custom no-PHI views are accompanied by changes to those views
        (including re-granting permission to the no-PHI user if need be)
  - [n/a] Liquibase rollback has been tested locally using `./gradlew liquibaseRollbackSQL` or `liquibaseRollback`
- [n/a] GraphQL schema changes are backward compatible with older version of the front-end

### Security
- [n/a] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [n/a] Any dependencies introduced have been vetted and discussed

## Cloud
- [n/a] DevOps team has been notified if PR requires ops support
- [n/a] If there are changes that cannot be tested locally, this has been deployed to our Azure `test` environment for verification

### CODE REVIEW

File: frontend/src/app/patients/personSchema.ts [hunk 1]
```diff
@@ -110,104 +117,140 @@ export function isValidBirthdate(date: string | undefined) {
...
+
+export type PersonErrors = Partial<Record<keyof PersonFormData, string>>;
 
 export type PhoneNumberErrors = Partial<Record<keyof PhoneNumber, string>>;
 
-export const getAllPhoneNumberErrors = (t: TFunction) => {
+export const usePersonSchemata = () => {
+  const { t } = useTranslation();
+  // TODO: translate this default error
+  const defaultValidationError = "Field is missing or invalid";
```
Comment (nickscialli-usds) [inline frontend/src/app/patients/personSchema.ts:245]: This needs translation but it _shouldn't_ ever be shown

File: frontend/src/app/Settings/Facility/facilitySchema.ts [hunk 2]
```diff
@@ -44,48 +71,60 @@ const providerSchema: yup.SchemaOf<RequiredProviderFields> = yup.object({
...
-      if (!input) {
-        return false;
+    .test(
+      "facility-phone",
+      "Facility phone number is missing or invalid",
+      function (input) {
+        if (!input) {
+          return false;
+        }
+        const number = phoneUtil.parseAndKeepRawInput(input, "US");
```
Comment (nathancrtr) [inline frontend/src/app/Settings/Facility/facilitySchema.ts:111]: You wrapped this function call in a try/catch in the person schema file, which was smart - want to do that here as well?

**Review by nullflux — APPROVED** [review]
This LGTM! Thanks for correcting the typing and adding those tests

File: frontend/src/app/Settings/Facility/facilitySchema.ts (hunk 2 above)
Comment (nickscialli-usds) [reply frontend/src/app/Settings/Facility/facilitySchema.ts]: 👍 implemented in fb8d68c

**Review by nathancrtr — APPROVED** [review]
Looks good! This anticipates several issues brought up during the bug hut and smoothes out some tricky spots with the recent i18n work. As mentioned by the author, some of the hooks are a little hairy, but this puts us in a good place to potentially explore other options on that front.

**Pull request merged**